docs/docs/sample-scripts/spo/create-test-documents-with-openai-api/index.md

```diff
@@ -0,0 +1,119 @@
+---
+tags:
+  - files
+  - libraries
+  - ai
+---
+
+# Create test documents in SharePoint with OpenAI API
+
+Author: [Nanddeep Nachan](https://nanddeepnachanblogs.com/posts/2023-02-12-test-docs-spo-openai-power-Automate/)
+
+While working with SharePoint, we most times need to create test content related to a specific topic. OpenAI API is the best fit in this case to generate content related to any topic.
+
+The below script shows how OpenAI API can be combined with CLI for Microsoft 365 to generate test documents in SharePoint.
+
+Prerequisites:
+- API Key is generated from the OpenAI. 
+- Document library should be available in the destination SharePoint site.
+
+=== "PowerShell"
+
+    ```powershell
+    param
+    (
+        [Parameter(Mandatory = $true, HelpMessage="The URL of the site where the files should be uploaded to")][string] $SiteURL,
+        [Parameter(Mandatory = $true, HelpMessage="Site-relative or server-relative URL to the folder where the files should be uploaded")][string] $Folder,
+        [Parameter(Mandatory = $true, HelpMessage="The OpenAI API key")][string] $OpenAIAPIKey,
+        [Parameter(Mandatory = $true, HelpMessage="Number of test files to generate")][int] $NumberOfDocuments
+    )
+
+    function New-WordFile {
+      param(
+          [string]$FileName,
+          [string]$Content
+      )
+
+      # Load Word application
+      $word = New-Object -ComObject Word.Application
+
+      # Create new document
+      $doc = $word.Documents.Add()
+
+      # Add content to document
+      $selection = $word.Selection
+      $selection.TypeText($Content)
+
+      # Save document
+      $path = Join-Path (Get-Location).Path $FileName
+      $doc.SaveAs("$path")
+      $doc.Close()
+
+      # Close Word application
+      $word.Quit()
+
+      # Clean up
+      $null = [System.Runtime.InteropServices.Marshal]::ReleaseComObject([System.__ComObject]$word)
+      [gc]::Collect()
+      [gc]::WaitForPendingFinalizers()
+      Remove-Variable word
+    }
+
+    try {
+      $m365Status = m365 status
+      if ($m365Status -eq "Logged Out") {
+        Write-Host "Logging in the User!"
+        m365 login --authType browser
+      }
+
+      $apiEndpoint = "https://api.openai.com/v1/completions"
+
+      $headers = @{
+        "Content-Type" = "application/json"
+        "Authorization" = "Bearer $OpenAIAPIKey"
+      }
+
+      $requestBody = @{
+        prompt = "Generate $NumberOfDocuments comma-separated distinct random noun words"
```

Comment:
Noticed that the AI was returning a csv with spaces in it that made our logs look ugly with double spaces. Rephrased the prompt so we're not getting a csv with spaces in it.